std/typecons.d

```diff
@@ -2679,7 +2684,11 @@ Params:
 
     size_t toHash() const @safe nothrow
```

Comment:
Two questions pop in my mind:
- shouldn't it infer everything, including safety and thrownness?
- if we are enforcing `@safe nothrow`, why not go all the way and add `pure @nogc`? In the end, computing the hash must not produce any garbage and should always have the same behaviour.

Reply:
That would make this incompatible with many objects since the Object.toHash isn't `pure` or `@nogc`. It would also prevent this from working with TypeInfo because TypeInfo.getHash isn't `pure` or `@nogc`.

Reply:
Cool. So then, imho this PR looks good. Probably in the future we could make an attempt to make `Object.toHash` and `TypeInfo.getHash` also `pure @nogc`.